File: solutions/julia/rna-transcription/3/rna-transcription.jl
nucleotide_dict = Dict('A'=>'U', 'T'=>'A', 'C'=>'G', 'G'=>'C') 

function is_valid(dna)
    match(r"[^ACGT]",dna) == nothing
end


function to_rna(dna)
    rna=""
    if is_valid(dna) 
        rna = join(map(x->nucleotide_dict[x], dna))
    else
       throw(error("Invalid DNA input.")) 
    end
    rna
end

File: solutions/clojure/rna-transcription/3/src/rna_transcription.clj
(ns rna-transcription)

(defn is-invalid? [dna]
  (some? (re-find #"(?=[BD-FH-S-Za-z0-9])" dna))
)

(defn to-rna [dna]  
    (if (is-invalid? dna)
      (throw (AssertionError.))
      (apply str (map {\G \C \C \G \T \A \A \U} dna))
      )    
 )